addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/ExternalRedirectScanRule.java

```diff
@@ -223,7 +229,10 @@ public void scan(HttpMessage msg, String param, String value) {
                 return;
             }
 
-            payload = REDIRECT_TARGETS[h];
+            payload =
+                    REDIRECT_TARGETS[h].contains(ORIGINAL_VALUE_PLACEHOLDER)
```

Comment:
At some point we should change this to enums and do the logic internally (with a boolean or null check), to not have to do contains over and over, knowing which ones need the replacement already.

Reply:
Do you mean something similar to what was done in https://github.com/zaproxy/zap-extensions/pull/6373 ?

Reply:
Yeah, it would be similar to that.